test/result-filter-output.js

```diff
@@ -0,0 +1,516 @@
+      publicMessages: [Message!] @filterData (disallow: "no-read-notes")
+    }
+    `,
+    resolvers: {
+      Query: {
+        publicMessages: async (parent, args, context, info) => {
+          return messages
+        }
+      }
+    }
+  })
+
+  app.register(mercuriusAuth, {
+    authContext: hasPermissionContext,
+    applyPolicy: hasFilterPolicy,
+    outputPolicyErrors: {
+      enabled: false
+    },
+    filterSchema: true,
+    authDirective: 'filterData'
+  })
+  try {
+    await app.inject({
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json',
+        'x-permission': 'no-read-notes'
+      },
+      url: '/graphql',
+      body: JSON.stringify({ query })
+    })
+  } catch (error) {
+    t.same(error, new MER_AUTH_ERR_USAGE_ERROR('Replacement can not happen on a resolver. Only a field.'))
+  }
+})
+
+test("ensure that a user who doesn't have the role to filter, still sees the notes as they normally would", async (t) => {
+  const app = Fastify()
+  t.teardown(app.close.bind(app))
+
+  app.register(mercurius, {
+    schema: `
+    directive @filterData (disallow: String!) on FIELD_DEFINITION
+
+    type Message {
+      message: String!
+      notes: String! @filterData (disallow: "no-read-notes")
+    }
+    
+    type Query {
+      publicMessages: [Message!]
+    }
+    `,
+    resolvers: {
+      Query: {
+        publicMessages: async (parent, args, context, info) => {
+          return messages
+        }
+      }
+    }
+  })
+
+  app.register(mercuriusAuth, {
+    authContext: hasPermissionContext,
+    applyPolicy: hasFilterPolicy,
+    outputPolicyErrors: {
+      enabled: false
+    },
+    filterSchema: true,
+    authDirective: 'filterData'
+  })
+
+  const response = await app.inject({
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+      'x-permission': 'read-notes'
+    },
+    url: '/graphql',
+    body: JSON.stringify({ query })
+  })
+
+  const { data } = JSON.parse(response.body)
+
+  t.plan(data.publicMessages.length)
+  for (let i = 0; i < data.publicMessages.length; i++) {
+    t.ok((data.publicMessages[i].notes !== null), 'notes are valid')
+  }
+})
+
+test('remove valid notes results and replace it with "foo" without any errors, using straight replace', async (t) => {
+  const app = Fastify()
+  t.teardown(app.close.bind(app))
+
+  app.register(mercurius, {
+    schema: `
+    directive @filterData (disallow: String!) on FIELD_DEFINITION
+
+    type Message {
+      message: String!
+      notes: String! @filterData (disallow: "no-read-notes")
+    }
+    
+    type Query {
+      publicMessages: [Message!]
+    }
+    `,
+    resolvers: {
+      Query: {
+        publicMessages: async (parent, args, context, info) => {
+          return messages
+        }
+      }
+    }
+  })
+
+  app.register(mercuriusAuth, {
+    authContext: hasPermissionContext,
+    applyPolicy: hasFilterPolicy,
+    outputPolicyErrors: {
+      enabled: false,
```

Comment:
Could we have a test for `enabled: true` + `valueOverride`as well?

Reply:
Done. Validation will fail during registration. Not sure else how to check for it since if we are not doing this block in the first place, we _aren't_ doing a replacement and errors wil be suprrused unless it's a schema error which would override this anyway. Seemed redundednt, but added. 👍